docs/design/messages.md

```diff
@@ -0,0 +1,89 @@
+# Gravity messages
```

Comment:
This is contained within the module/spec. When we go to render the doucmentation, the spec in the module will be used. Can we merge any information from here into there?

Reply:
Where is the spec being worked on? I'll compare them on review.